h/panels/navbar.py

```diff
@@ -29,14 +30,20 @@ def navbar(context, request, search=None, opts=None):
                 'title': group.name,
                 'link': request.route_url('group_read', pubid=group.pubid, slug=group.slug)
             })
-            groups_suggestions.append({
-                'name': group.name,
-                'pubid': group.pubid
-            })
         user_activity_url = request.route_url('activity.user_search',
                                               username=request.user.username)
         username = request.user.username
 
+    # make all groups visible in the search auto complete
+    groups = profile_groups_factory(None, request).all(request.user, request.authority)
+    for group in groups:
+        # don't return __world__ group
+        if group['id'] == '__world__':
```

Comment:
We can work together to figure this out as it comes up, but if this code ends up using the `ListGroupService` to query for groups, groups are going to come back as true `Group` model objects (with attributes), not lists of dicts as it is now. Formatting-as-dicts has been factored out of the service itself in lieu of a new presenter class (two, technically: `GroupJSONPresenter` and `GroupsJSONPresenter`)...so either we'll want to rejigger this to use group objects (most likely, I'd think?) or run the results through the presenter to get dicts (likely a useless extra step).

Reply:
Yes indeed! I'll update this to use the /groups api and we can take another peak at it tomorrow.